src/Features/Core/Portable/ConflictMarkerResolution/AbstractConflictMarkerCodeFixProvider.cs

```diff
@@ -45,91 +45,171 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+            if (position != conflictLine.Start)
+            {
+                Debug.Assert(false, "All conflict markers should start at the beginning of a line.");
+                return false;
+            }
+
+            if (!TryGetConflictLines(text, position, out startLine, out middleLine, out endLine))
+                return false;
+
+            var startTrivia = root.FindTrivia(startLine.Start);
+            var middleTrivia = root.FindTrivia(middleLine.Start);
+
+            if (position == middleLine.Start)
+            {
+                // we were on the ======= lines.  We only want to report here if there was no
+                // conflict trivia on the <<<<<<< line (since we would have already reported the
+                // issue there.
+                if (startTrivia.RawKind == _syntaxKinds.ConflictMarkerTrivia)
+                    return false;
+            }
+            else if (position == endLine.Start)
+            {
+                // we were on the >>>>>>> lines.  We only want to report here if there was no
+                // conflict trivia on the ======= or <<<<<<< line (since we would have already reported the
+                // issue there.
+                if (startTrivia.RawKind == _syntaxKinds.ConflictMarkerTrivia ||
+                    middleTrivia.RawKind == _syntaxKinds.ConflictMarkerTrivia)
+                    return false;
+            }
+
+            return true;
+        }
+
+        private bool TryGetConflictLines(
+            SourceText text, int position,
+            out TextLine startLine, out TextLine middleLine, out TextLine endLine)
+        {
+            startLine = default;
+            middleLine = default;
+            endLine = default;
+
+            var lines = text.Lines;
+            switch (text[position])
+            {
+                case '<':
+                    startLine = lines.GetLineFromPosition(position);
+                    return TryFindLineForwards(startLine, '=', out middleLine) &&
+                           TryFindLineForwards(middleLine, '>', out endLine);
+                case '=':
+                    middleLine = lines.GetLineFromPosition(position);
+                    return TryFindLineBackwards(middleLine, '<', out startLine) &&
+                           TryFindLineForwards(middleLine, '>', out endLine);
+                case '>':
+                    endLine = lines.GetLineFromPosition(position);
+                    return TryFindLineBackwards(endLine, '=', out middleLine) &&
+                           TryFindLineBackwards(middleLine, '<', out startLine);
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(text[position]);
+            }
+        }
+
+        private bool TryFindLineForwards(TextLine startLine, char ch, out TextLine foundLine)
+        {
+            var text = startLine.Text!;
+            var lines = text.Lines;
+            for (var i = startLine.LineNumber + 1; i < lines.Count; i++)
```

Comment:
startLine might be the last line

Reply:
note: in that case, this for-loop will bail out immediately.

Reply:
You are right. I was a little confused why there's a diagnostic even if not all markers exist. So, is the diagnostic created by compiler?

---
In reply to: [395393643](https://github.com/dotnet/roslyn/pull/42213#discussion_r395393643) [](ancestors = 395393643)

Reply:
added test.

Reply:
> You are right. I was a little confused why there's a diagnostic even if not all markers exist. So, is the diagnostic created by compiler?

Yes.  diagnsotic is created by compiler when tehy see `<<<<<<` and the like.  it becomes trivia (normally) and they issue a diagnostic.

This worked well initially for the codefix, but falls over in the case where the stuff is in string-literals.  So i'm fixing that up now.  We still need at least one of these to be in real code.  but as long as tehre's one, we can fix them up :)